lib/discourse_activity_pub/ap/activity/announce.rb

```diff
@@ -7,6 +7,20 @@ class Announce < Activity
         def type
           'Announce'
         end
+
+        # See further https://codeberg.org/fediverse/fep/src/branch/main/fep/1b12/fep-1b12.md#the-announce-activity
+        def process
+          @actor = Actor.resolve_and_store(json[:actor])
+          return process_failed("cant_create_actor") unless actor.present?
+
+          @object = Object.resolve_and_store(json[:object], self)
+          return process_failed("cant_find_object") unless object.present?
+          return false unless perform_validate_activity
+
+          object.parent = self
```

Comment:
I am seeing some errors on this line locally. I have set up a category to follow my account [pmusaraj@mstdn.ca](https://mstdn.ca/users/pmusaraj), and after a boosted post on that account, I see an error in Logster: 

```
Job exception: undefined method `parent=' for #<DiscourseActivityPub::AP::Object::Note:0x000000012cc0d1b8 @json={"@context"=>["https://www.w3.org/ns/activitystreams", {"ostatus"=>"http://ostatus.org#", "atomUri"=>"ostatus:atomUri", "inReplyToAtomUri"=>"ostatus:inReplyToAtomUri", "conversation"=>"ostatus:conversation", "sensitive"=>"as:sensitive", "toot"=>"http://joinmastodon.org/ns#", "votersCount"=>"toot:votersCount", "blurhash"=>"toot:blurhash", "focalPoint"=>{"@container"=>"@list", "@id"=>"toot:focalPoint"}}], "id"=>"https://hachyderm.io/users/lkanies/statuses/111568355721240341", "type"=>"Note", "summary"=>nil, "inReplyTo"=>nil, "published"=>"2023-12-12T16:20:06Z", "url"=>"https://hachyderm.io/@lkanies/111568355721240341", "attributedTo"=>"https://hachyderm.io/users/lkanies", "to"=>["https://www.w3.org/ns/activitystreams#Public"], "cc"=>["https://hachyderm.io/users/lkanies/followers"], "sensitive"=>false, "atomUri"=>"https://hachyderm.io/users/lkanies/statuses/111568355721240341", "inReplyToAtomUri"=>nil, "conversation"=>"tag:hachyderm.io,2023-12-12:objectId=107533779:objectType=Conversation", "content"=>"<p>A funny article about home decor and parenting.</p><p>Do interior designers know what homes are for? <a href=\"https://evilwitches.substack.com/p/do-interior-designers-know-what-homes\" target=\"_blank\" rel=\"nofollow noopener noreferrer\" translate=\"no\"><span class=\"invisible\">https://</span><span class=\"ellipsis\">evilwitches.substack.com/p/do-</span><span class=\"invisible\">interior-designers-know-what-homes</span></a></p>", "contentMap"=>{"en"=>"<p>A funny article about home decor and parenting.</p><p>Do interior designers know what homes are for? <a href=\"https://evilwitches.substack.com/p/do-interior-designers-know-what-homes\" target=\"_blank\" rel=\"nofollow noopener noreferrer\" translate=\"no\"><span class=\"invisible\">https://</span><span class=\"ellipsis\">evilwitches.substack.com/p/do-</span><span class=\"invisible\">interior-designers-...
```

Reply:
hm yes, I'll add a fix and test for that case tomorrow.

Reply:
@pmusaraj Ok, this was a great catch as it led me to properly disambiguating the Create actor and the Object attributedTo. https://github.com/angusmcleod/discourse-activity-pub/commit/a8ecd943e0be4f850989b5de3e325908753363ff

Reply:
Excellent, thanks. Will test in a few hours.